### PR TITLE
Update Python example to align with v3 API

### PR DESCRIPTION
**Pull request by joannasese**
- Examples use v3 API
- Replaces subscription create endpoint with purchase create

### CODE REVIEW

File: api/python/app.py [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
-import recurly
 
 # We'll use uuid to generate unique account codes
 import uuid
 
-# Configure the recurly client with your subdomain and api key
-recurly.SUBDOMAIN = os.environ['RECURLY_SUBDOMAIN']
+# Configure the recurly client with your api key
 recurly.API_KEY = os.environ['RECURLY_API_KEY']
+client = recurly.Client(recurly.API_KEY)
```
Comment (joannasese) [inline api/python/app.py:15]: `AttributeError: module 'recurly' has no attribute 'Client'`

File: api/python/app.py [hunk 2]
```diff
@@ -10,9 +10,9 @@
 # We'll use uuid to generate unique account codes
 import uuid
 
-# Configure the recurly client with your subdomain and api key
-recurly.SUBDOMAIN = os.environ['RECURLY_SUBDOMAIN']
+# Configure the recurly client with your api key
 recurly.API_KEY = os.environ['RECURLY_API_KEY']
```
Comment (douglasmiller) [inline api/python/app.py:14]: This (`recurly.API_KEY`) is a vestige of the v2 client. The below line should just use the environment variable directly.

**Comment by douglasmiller** [comment]
The v3 client does not have a [`recurly.BillingInfo`](https://github.com/recurly/recurly-integration-examples/pull/75/files#diff-5c5f28beefc76656c4a9f4ec6355a175R45). This should be a dict `{}` instead

File: api/python/app.py [hunk 3]
```diff
@@ -51,21 +51,26 @@ def new_subscription():
...
+        "subscriptions": [{"plan_code": "basic"}],
+    }
+    invoice_collection = client.create_purchase(purchase_create)
+
+    # The purchase has been created and we can redirect
     # to a confirmation page
-    subscription.save()
+
     return redirect(SUCCESS_URL)
   except recurly.ValidationError as error:
```
Comment (douglasmiller) [inline api/python/app.py:75]: The below 3d secure transaction error will not be thrown as a `ValidationError`. Instead, a `recurly.errors.TransactionError` will be thrown.

Comment (douglasmiller) [reply api/python/app.py:75]: A handler for the generic `recurly.ApiError` should follow the `TransactionError` handler for all other possible errors with the `error_redirect` call.

File: api/python/app.py [hunk 4]
```diff
@@ -84,13 +89,24 @@ def new_subscription():
...
+
+    account_create = {
+      "code": recurly_account_code,
+      "first_name": request.form['first-name'],
+      "last_name": request.form['last-name'],
+      "billing_info": billing_info
+    }
+    account = client.create_account(account_create)
     return redirect(SUCCESS_URL)
   except recurly.ValidationError as errors:
```
Comment (douglasmiller) [inline api/python/app.py:111]: `recurly.ValidationError` is the v2 client naming scheme. The v3 clients use `recurly.errors.ValidationError`

**Comment by douglasmiller** [comment]
In general, reference the correct [error namespace](https://github.com/recurly/recurly-client-python/blob/master/recurly/errors.py) for the `except` blocks

**Review by chrissrogers — APPROVED** [review]

**Pull request merged**